src/uct/cuda/cuda_ipc/cuda_ipc_iface.c

```diff
@@ -107,11 +107,20 @@ uct_cuda_ipc_iface_is_reachable_v2(const uct_iface_h tl_iface,
     uct_cuda_ipc_md_t *md        = ucs_derived_of(base_iface->md, uct_cuda_ipc_md_t);
 #endif
 
-    if (!uct_iface_is_reachable_params_addrs_valid(params) ||
-        (getpid() == *(pid_t*)params->iface_addr) ||
-        !uct_iface_scope_is_reachable(tl_iface, params)) {
+    if (!uct_iface_is_reachable_params_addrs_valid(params)) {
         return 0;
-    };
+    }
+
+    if (getpid() == *(pid_t*)params->iface_addr) {
+        uct_iface_fill_info_str_buf(params,
+                                    "different pid %d vs %d", getpid(),
+                                    *(pid_t*)params->iface_addr);
+        return 0;
+    }
+
+    if (!uct_iface_scope_is_reachable(tl_iface, params)) {
```

Comment:
maybe unite this with the 'if' in line 110

Reply:
It means switching the order between checking the process id and calling uct_iface_scope_is_reachable (which does more checks). Do you think it's ok?

Reply:
actually, i think we need to fix an issue introduced by another PR and move uct_iface_scope_is_reachable to the end of the function (return uct_iface_scope_is_reachable instead of returning 1) because we want to call uct_iface_scope_is_reachable only after making sure the address is reachable